test/types/tuple/types/TupleTest.chpl

```diff
@@ -1,9 +1,18 @@
 proc test(type t) where isTuple(t) {
   var a = ((new t[0](1), new t[1](2)), (new t[0](3), new t[1](4)));
   assert(a.size == 2);
+  if isUnmanagedClass(t) {
+    delete a;
+  }
 }
 
 proc test(type t) {
   var a = (new t(1), new t(2));
   assert(a.size == 2);
+
+  if isUnmanagedClass(t) {
+    for i in a.indices {
+      delete a[i];
```

Comment:
I am a bit confused here. Above we delete a tuple of tuples and it deletes all the way down, but here we delete indices of the tuple one by one.

As long as it works and closes the leak, I don't have an issue with this. But I don't know what's the right way to cleanup a tuple of unmanageds (or tuple of tuple of unmanageds)

Reply:
You should be confused, because the above version is always param-folded out, since we only test `shared` tuples. Removing the above version soon.